internal/cmd/logout.go

```diff
@@ -83,92 +98,43 @@ func logout(clear bool, server string, all bool) error {
 
 	switch {
 	case all:
-		stateChanged := false
+		logging.S.Debug("logging out of all servers\n")
+	case server == "":
+		logging.S.Debug("logging out of current server\n")
+	default:
+		logging.S.Debugf("logging out of server [%s]\n", server)
+	}
+
+	if all {
 		for i := range config.Hosts {
-			if config.Hosts[i].isLoggedIn() {
-				logoutOfServer(&config.Hosts[i])
-				stateChanged = true
-			}
-		}
-		if stateChanged {
-			fmt.Fprintf(os.Stderr, "  Logged out of all servers.\n")
-		} else {
-			fmt.Fprintf(os.Stderr, "  Not logged in to any servers.\n")
+			logoutOfServer(&config.Hosts[i])
 		}
-
+		fmt.Fprintf(os.Stderr, "Logged out of all servers.\n")
 		if clear {
-			if config.Hosts != nil {
-				config.Hosts = nil
-				fmt.Fprintf(os.Stderr, "  Cleared list of all servers\n")
-			} else {
-				fmt.Fprintf(os.Stderr, "  No servers to clear.\n")
-			}
+			config.Hosts = nil
+			logging.S.Debug("cleared all servers from login list\n")
 		}
-	case server == "":
-		serverFound := false
+	} else {
 		for i := range config.Hosts {
-			if config.Hosts[i].Current {
-				serverFound = true
-				if config.Hosts[i].isLoggedIn() {
-					logoutOfServer(&config.Hosts[i])
-					fmt.Fprintf(os.Stderr, "  Logged out of server %s.\n", config.Hosts[i].Host)
-				} else {
-					fmt.Fprintf(os.Stderr, "  Not logged in to server %s.\n", config.Hosts[i].Host)
+			if (server == "" && config.Hosts[i].Current) || (server == config.Hosts[i].Host) {
+				success := logoutOfServer(&config.Hosts[i])
+				if success {
+					logging.S.Debugf("Logged out of server %s", config.Hosts[i].Host)
 				}
-				break
-			}
-		}
-		if serverFound {
-			if clear {
-				var newHostConfigs []ClientHostConfig
 
-				// New slice of hosts except the one to clear
-				for i := range config.Hosts {
-					if config.Hosts[i].Current {
-						fmt.Fprintf(os.Stderr, "  Cleared [%s] from list of servers\n", config.Hosts[i].Host)
-						continue
+				if clear {
+					serverUrl := config.Hosts[i].Host
+					if len(config.Hosts) < 2 {
```

Comment:
Is there a way to just make it an empty slice and avoid this `if` branch?

Reply:
we could, but that requires more refactoring because it triggers a different error flow in a bunch of places (ie. we just check if its nil rather than len=0 and error out if so)